setup.py

```diff
@@ -781,16 +781,24 @@ def build_torch_extension(build_ext, options, torch_version):
         pytorch_lib.__dict__[k] = v
     build_ext.build_extension(pytorch_lib)
 
+def execute_cmd(cmd):
+    make_process = subprocess.Popen(cmd, stdout=sys.stdout, stderr=sys.stderr, shell=True)
+    make_process.communicate()
+    if make_process.returncode:
+        raise DistutilsSetupError('An ERROR occured while running the command: \n' +
+            cmd + '\n Exit code: {0}'.format(make_process.returncode))
 
 # run the customize_compiler
 class custom_build_ext(build_ext):
     def build_extensions(self):
+        execute_cmd(os.environ['BYTEPS_PRE_SETUP_SCRIPT'])
```

Comment:
Wouldn't `os.environ['BYTEPS_PRE_SETUP_SCRIPT']` throw exception if there is no such env variable? Can you use `os.environ.get()` with a default value to avoid that?